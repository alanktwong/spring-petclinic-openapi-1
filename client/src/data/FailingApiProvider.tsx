import * as React from 'react';
import { FailingApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithFailingApiProps {
  failingApi: FailingApi;
}

export const FailingApiContext = React.createContext(new FailingApi());

export const withFailingApi = <P extends WithFailingApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithFailingApi extends React.Component<
    Subtract<P, WithFailingApiProps>
  > {
    static contextType = FailingApiContext;
    context!: React.ContextType<typeof FailingApiContext>;

    render() {
      return <Component failingApi={this.context} {...(this.props as P)} />;
    }
  };
