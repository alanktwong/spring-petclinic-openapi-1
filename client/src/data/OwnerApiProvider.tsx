import * as React from 'react';
import { OwnerApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithOwnerApiProps {
  ownerApi: OwnerApi;
}

export const OwnerApiContext = React.createContext(new OwnerApi());

export const withOwnerApi = <P extends WithOwnerApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithOwnerApi extends React.Component<Subtract<P, WithOwnerApiProps>> {
    static contextType = OwnerApiContext;
    context!: React.ContextType<typeof OwnerApiContext>;

    render() {
      return <Component ownerApi={this.context} {...(this.props as P)} />;
    }
  };
