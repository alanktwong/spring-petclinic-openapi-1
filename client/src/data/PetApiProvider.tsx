import * as React from 'react';
import { PetApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithPetApiProps {
  petApi: PetApi;
}

export const PetApiContext = React.createContext(new PetApi());

export const withPetApi = <P extends WithPetApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithPetApi extends React.Component<Subtract<P, WithPetApiProps>> {
    static contextType = PetApiContext;
    context!: React.ContextType<typeof PetApiContext>;

    render() {
      return <Component petApi={this.context} {...(this.props as P)} />;
    }
  };
