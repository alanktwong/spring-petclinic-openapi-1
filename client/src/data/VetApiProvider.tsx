import * as React from 'react';
import { VetApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithVetApiProps {
  vetApi: VetApi;
}

export const VetApiContext = React.createContext(new VetApi());

export const withVetApi = <P extends WithVetApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithVetApi extends React.Component<Subtract<P, WithVetApiProps>> {
    static contextType = VetApiContext;
    context!: React.ContextType<typeof VetApiContext>;

    render() {
      return <Component vetApi={this.context} {...(this.props as P)} />;
    }
  };
