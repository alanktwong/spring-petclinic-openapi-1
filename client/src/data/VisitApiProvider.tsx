import * as React from 'react';
import { VisitApi } from 'petclinic-api';
import { Subtract } from 'utility-types';

export interface WithVisitApiProps {
  visitApi: VisitApi;
}

export const VisitApiContext = React.createContext(new VisitApi());

export const withVisitApi = <P extends WithVisitApiProps>(
  Component: React.ComponentType<P>
) =>
  class WithVisitApi extends React.Component<Subtract<P, WithVisitApiProps>> {
    static contextType = VisitApiContext;
    context!: React.ContextType<typeof VisitApiContext>;

    render() {
      return <Component visitApi={this.context} {...(this.props as P)} />;
    }
  };
