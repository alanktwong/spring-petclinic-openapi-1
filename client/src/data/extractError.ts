import { IError } from '../types';

const extractError = (response): Promise<IError> => {
  if (response.json) {
    return response.json();
  } else {
    return Promise.resolve({ message: `${response}` });
  }
};

export default extractError;
