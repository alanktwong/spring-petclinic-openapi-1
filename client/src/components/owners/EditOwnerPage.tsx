import * as React from 'react';

import { RouteComponentProps, withRouter } from 'react-router-dom';

import OwnerEditor from './OwnerEditor';

import { Owner } from 'petclinic-api';
import { withOwnerApi, WithOwnerApiProps } from '../../data/OwnerApiProvider';

import { IError } from '../../types';
import PageErrorMessage from '../PageErrorMessage';
import extractError from '../../data/extractError';

interface IEditOwnerPageProps extends RouteComponentProps, WithOwnerApiProps {}

interface IEditOwnerPageState {
  error?: IError;
  owner?: Owner;
}

class EditOwnerPage extends React.Component<
  IEditOwnerPageProps,
  IEditOwnerPageState
> {
  async componentDidMount() {
    const ownerId = Number(this.props.match.params['ownerId']);

    try {
      const owner = await this.props.ownerApi.getOwner({ ownerId });
      this.setState({ owner });
    } catch (response) {
      const error = await extractError(response);
      this.setState({ error });
    }
  }

  render() {
    const { error, owner } = this.state || {};

    if (error) {
      return <PageErrorMessage error={error} />;
    }

    if (!owner) {
      return <div>Loading...</div>;
    }

    return <OwnerEditor initialOwner={owner} />;
  }
}

export default withOwnerApi(withRouter(EditOwnerPage));
