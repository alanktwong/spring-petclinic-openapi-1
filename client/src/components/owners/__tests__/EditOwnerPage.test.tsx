import React from 'react';
import renderer from 'react-test-renderer';
import EditOwnerPage from '../EditOwnerPage';
import waitForExpect from 'wait-for-expect';
import { OwnerApiContext } from '../../../data/OwnerApiProvider';
import { OwnerApi } from 'petclinic-api';
import { Route, StaticRouter } from 'react-router-dom';

it('renders correctly', async () => {
  const mock = new OwnerApi();
  mock.getOwner = async () => ({
    id: 1,
    firstName: 'firstName1',
    lastName: 'lastName1',
    address: 'address1',
    city: 'city1',
    telephone: '012345',
    pets: [],
  });

  const component = renderer.create(
    <StaticRouter location="/owners/1/edit">
      <OwnerApiContext.Provider value={mock}>
        <Route path="/owners/:ownerId(\d+)/edit" component={EditOwnerPage} />
      </OwnerApiContext.Provider>
    </StaticRouter>
  );

  const page = component.root.find(node => node.type.name === 'EditOwnerPage');

  await waitForExpect(() => {
    expect(page.instance.state.owner.id).toBe(1);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
