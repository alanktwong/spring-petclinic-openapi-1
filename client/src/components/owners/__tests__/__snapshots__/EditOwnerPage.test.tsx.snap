// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2>
    New Owner
  </h2>
  <form
    className="form-horizontal"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="form-group has-feedback"
    >
      <div
        className="form-group "
      >
        <label
          className="col-sm-2 control-label"
          htmlFor="firstName"
        >
          First Name
        </label>
        <div
          className="col-sm-10"
        >
          <input
            className="form-control"
            id="firstName"
            name="firstName"
            onBlur={[Function]}
            onChange={[Function]}
            value="firstName1"
          />
          <span
            aria-hidden="true"
            className="glyphicon glyphicon-ok form-control-feedback"
          />
        </div>
      </div>
      <div
        className="form-group "
      >
        <label
          className="col-sm-2 control-label"
          htmlFor="lastName"
        >
          Last Name
        </label>
        <div
          className="col-sm-10"
        >
          <input
            className="form-control"
            id="lastName"
            name="lastName"
            onBlur={[Function]}
            onChange={[Function]}
            value="lastName1"
          />
          <span
            aria-hidden="true"
            className="glyphicon glyphicon-ok form-control-feedback"
          />
        </div>
      </div>
      <div
        className="form-group "
      >
        <label
          className="col-sm-2 control-label"
          htmlFor="address"
        >
          Address
        </label>
        <div
          className="col-sm-10"
        >
          <input
            className="form-control"
            id="address"
            name="address"
            onBlur={[Function]}
            onChange={[Function]}
            value="address1"
          />
          <span
            aria-hidden="true"
            className="glyphicon glyphicon-ok form-control-feedback"
          />
        </div>
      </div>
      <div
        className="form-group "
      >
        <label
          className="col-sm-2 control-label"
          htmlFor="city"
        >
          City
        </label>
        <div
          className="col-sm-10"
        >
          <input
            className="form-control"
            id="city"
            name="city"
            onBlur={[Function]}
            onChange={[Function]}
            value="city1"
          />
          <span
            aria-hidden="true"
            className="glyphicon glyphicon-ok form-control-feedback"
          />
        </div>
      </div>
      <div
        className="form-group "
      >
        <label
          className="col-sm-2 control-label"
          htmlFor="telephone"
        >
          Telephone
        </label>
        <div
          className="col-sm-10"
        >
          <input
            className="form-control"
            id="telephone"
            name="telephone"
            onBlur={[Function]}
            onChange={[Function]}
            value="012345"
          />
          <span
            aria-hidden="true"
            className="glyphicon glyphicon-ok form-control-feedback"
          />
        </div>
      </div>
    </div>
    <div
      className="form-group"
    >
      <div
        className="col-sm-offset-2 col-sm-10"
      >
        <button
          className="btn btn-default"
          disabled={false}
          type="submit"
        >
          Update Owner
        </button>
      </div>
    </div>
  </form>
</div>
`;
